### PR TITLE
Skip parameters which are not there instead of failing out

### LINKED ISSUE #77: Be more robust against malformed parameters
See https://github.com/nextcloud/server/issues/2255

### LINKED ISSUE #256: Strange behaviors of Nextcloud in Activity app
[Here more information](https://help.nextcloud.com/t/strange-behaviors-of-nextcloud-in-activity-app/29082)

### Steps to reproduce
1. share file by email address in web interface
2. activity app don't show the activities prior to sharing
3.

### Expected behaviour
Activity app should show all previous activities and also sharing one

### Actual behaviour
When I share a public link with email the activity app (from web interface) does not show me the activity of sharing nor previous activities letting the icon of waiting forever run.

### Server configuration

**Operating system**: 
Ubuntu server 17.10 x64

**Web server:**
apache2-2.4.27

**Database:**
mysql-5.7.21

**PHP version:**
php-7.1

**Nextcloud version:**
13.0.1.1

**Where did you install Nextcloud from:**
Update from 13.0.0.14 (but with same issue)

**Signing status:**
No errors have been found.

**List of activated apps:**
Enabled:
  - activity: 2.6.1
  - admin_audit: 1.3.0
  - bruteforcesettings: 1.0.3
  - calendar: 1.6.1
  - comments: 1.3.0
  - contacts: 2.1.2
  - dav: 1.4.6
  - federatedfilesharing: 1.3.1
  - files: 1.8.0
  - files_downloadactivity: 1.2.0
  - files_external: 1.4.1
  - files_pdfviewer: 1.2.0
  - files_sharing: 1.5.0
  - files_texteditor: 2.5.1
  - files_trashbin: 1.3.0
  - files_versions: 1.6.0
  - files_videoplayer: 1.2.0
  - firstrunwizard: 2.2.1
  - gallery: 18.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.1.0
  - mail: 0.7.10
  - nextcloud_announcements: 1.2.0
  - notifications: 2.1.2
  - oauth2: 1.1.0
  - password_policy: 1.3.0
  - phonetrack: 0.2.2
  - provisioning_api: 1.3.0
  - quota_warning: 1.2.0
  - serverinfo: 1.3.0
  - sharebymail: 1.3.0
  - spreed: 3.1.0
  - survey_client: 1.1.0
  - systemtags: 1.3.0
  - theming: 1.4.1
  - twofactor_backupcodes: 1.2.3
  - updatenotification: 1.3.0
  - workflowengine: 1.3.0
Disabled:
  - encryption
  - federation
  - user_external
  - user_ldap

**Nextcloud configuration:**
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "xxx.yyy.zzz"
        ],
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "filelocking.enabled": "true",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0
        },
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "https:\/\/xxx.yyy.zzz",
        "htaccess.RewriteBase": "\/",
        "dbtype": "mysql",
        "version": "13.0.1.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "log_type": "nextcloud",
        "logfile": "nextcloud.log",
        "loglevel": 2,
        "logtimezone": "Europe\/Rome",
        "logdateformat": "F d, Y H:i:s",
        "mail_smtpmode": "sendmail",
        "mail_smtpauthtype": "PLAIN",
        "maintenance": false,
        "theme": "",
        "auth.bruteforce.protection.enabled": true,
        "trashbin_retention_obligation": "auto",
        "versions_retention_obligation": "auto",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***"
    }
}

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No

### Client configuration
**Browser:**
Firefox 58
**Operating system:**
win10 64bit

#### Nextcloud log (data/nextcloud.log)
none

#### Browser log
none

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #256 
Fix #77 

### Testing today:
1. Switch to italian
2. Create a file
3. Share by email

### Testing tomorrow:
See above + revert https://github.com/nextcloud/activity/issues/256#issuecomment-374560208

### Expected
1. The stream is loaded (broken parameters displayed as is)

### Actual
1. The stream breaks completly.

### CODE REVIEW

File: js/richObjectStringParser.js [hunk 1]
```diff
@@ -45,8 +45,12 @@
 
 			_.each(matches, function(parameter) {
 				parameter = parameter.substring(1, parameter.length - 1);
-				var parsed = self.parseParameter(parameters[parameter]);
+				if (!parameters.hasOwnProperty(parameter) || !parameters[parameter]) {
+					// Malformed translation?
```
Comment (blizzz) [inline js/richObjectStringParser.js]: should this be reported instead of failing silently?

**Comment by blizzz** [comment]
travis is unhapy

File: js/richObjectStringParser.js (hunk 1 above)
Comment (nickvergessen) [reply js/richObjectStringParser.js]: Well only thing would be  console.log, but that is disabled for non-debug anyway, so I don't think it's important.

**Comment by nickvergessen** [comment]
Travis was fixed with https://github.com/nextcloud/activity/pull/259/files

But I will rebase

File: js/richObjectStringParser.js (hunk 1 above)
Comment (blizzz) [reply js/richObjectStringParser.js]: so, wouldn't it be helpful for debugging? for the app dev to to find out where an issues occurs?

Comment (nickvergessen) [reply js/richObjectStringParser.js]: whatever I was thinking, yes of course, amended

**Comment by MorrisJobke** [comment]
@nickvergessen Next RC is coming - time for backport

**Comment by MorrisJobke** [comment]
@nickvergessen To backport or not to backport?

**Comment by nickvergessen** [comment]
ws already backported see #265

**Pull request merged**